fix!: output-module-groups to work like run-all

bt-macole opened the pull request:
<!-- Prepend '[WIP]' to the title if this PR is still a work-in-progress. Remove it when it is ready for review! -->

## Description

output-module-groups was not working from a directory that did not have a terragrunt.hcl resource and it was not working with terragrunt cli options such as `terragrunt-modules-that-include`.

furthermore, it was not inverting order for the destroy command.

and required init even though it is just using the stackconfig to output planned behavior.

this change aligns the behavior with run-all so that it will work with terragrunt cli options, properly display order based on apply and/or destory, and can be run within any folder that run-all could be executed in.

Fixes #2685
Fixes #2768

BREAKING CHANGE: technically this does change the behavior, in that it no longer requires terraform init. but i believe that more accurately reflects the desired behavior.


<!-- Description of the changes introduced by this PR. -->

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [x] Update the docs -- based on conventions with sub commands, no additional documentation seemed required.
- [x] Run the relevant tests successfully, including pre-commit checks.
- [x] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [x] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

<!-- One-line description of the PR that can be included in the final release notes. -->
Added / Removed / Updated [X].

Updated `output-module-groups` to work like run-all from any sub folder within a terragrunt configuration
Updated `output-module-groups` no longer requires `init` which greatly improves performance
Added support for terragrunt cli options for `output-module-groups`
Added support for `destroy` sub command for `output-module-groups` to generate reversed dependency ordered stack list

### Migration Guide

<!-- Important: If you made any backward incompatible changes, then you must write a migration guide! -->

If any manual or automated process relied on the modules being initialized with `init` when running `output-module-groups` those processes will need to be updated to `init` via some other mechanism such as `terragrunt run-all init`

bt-macole commented:
just noticed d5204846b35e4c8f45e51bbf7541c1430e70a7a4 has the same change to cli/commands/output-module-groups/action.go, happy to cherry-pick that here or visa versa.

evsl commented:
> just noticed #2769 has the same change to cli/commands/output-module-groups/action.go ([d520484](https://github.com/gruntwork-io/terragrunt/commit/d5204846b35e4c8f45e51bbf7541c1430e70a7a4)), happy to cherry-pick that here or visa versa.

That would be neat! I was just poking about to consider updating my PR based on those changes, and I would certainly imagine that @denis256 would also like an integration test around the new functionality.

Additionally, relating to tests, my PR does fix the existing integration test for `output-module-groups` (it already existed, but it was erroneously passing due to the presence of a terragrunt.hcl file in the root of the test fixture, which I simply moved).

denis256 commented on cli/commands/output-module-groups/command.go:
```diff
@@ -6,13 +6,42 @@ import (
+func subCommandFunc(cmd string, opts *options.TerragruntOptions) *cli.Command {
+	opts.TerraformCommand = cmd
+	return &cli.Command{
+		Name:   cmd,
```
will be helpful to define `Usage` to show how subcommands can be used

![image](https://github.com/gruntwork-io/terragrunt/assets/10694338/e1378b23-5391-4bf7-98f1-2241e2546ce1)

bt-macole replied:
I originally had usage for the sub-commands but felt it was repetitive. updated!

denis256 reviewed (commented):
Hello,
will be helpful to cherry-pick all changes in one PR, which looks like was already applied for `cli/commands/output-module-groups/action.go`, remaining only to remove `terragrunt.hcl`

PR misses the documentation page to show examples of how to use `output-module-groups` subcommands

bt-macole commented:
> Hello,
> will be helpful to cherry-pick all changes in one PR, which looks like was already applied for cli/commands/output-module-groups/action.go, remaining only to remove terragrunt.hcl
>
> PR misses the documentation page to show examples of how to use output-module-groups subcommands

updated. also added table driven test to cover all subcommands

bt-macole commented:
@evsl good stuff!

thanks @denis256 🎉
